core/modules/replica/FindRequest.cc

```diff
@@ -0,0 +1,308 @@
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "startImpl "
+         << " worker: "          << worker()
+         << " database: "        << database()
+         << " chunk: "           << chunk()
+         << " computeCheckSum: " << (computeCheckSum() ? "true" : "false"));
+
+    // Serialize the Request message header and the request itself into
+    // the network buffer.
+
+    _bufferPtr->resize();
+
+    proto::ReplicationRequestHeader hdr;
+    hdr.set_id(id());
+    hdr.set_type(proto::ReplicationRequestHeader::REPLICA);
+    hdr.set_replica_type(proto::ReplicationReplicaRequestType::REPLICA_FIND);
+
+    _bufferPtr->serialize(hdr);
+
+    proto::ReplicationRequestFind message;
+    message.set_priority(priority());
+    message.set_database(database());
+    message.set_chunk(chunk());
+    message.set_compute_cs(computeCheckSum());
+
+    _bufferPtr->serialize(message);
+
+    send();
+}
+
+void FindRequestM::wait() {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "wait");
+
+    // Allways need to set the interval before launching the timer.
+
+    _timer.expires_from_now(boost::posix_time::seconds(_timerIvalSec));
+    _timer.async_wait(
+        boost::bind(
+            &FindRequestM::awaken,
+            shared_from_base<FindRequestM>(),
+            boost::asio::placeholders::error
+        )
+    );
+}
+
+void FindRequestM::awaken(boost::system::error_code const& ec) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "awaken");
+
+    if (isAborted(ec)) { return; }
+
+    // Also ignore this event if the request expired
+    if (_state== State::FINISHED){ return; }
+
+    LOCK_GUARD;
+
+    // Serialize the Status message header and the request itself into
+    // the network buffer.
+
+    _bufferPtr->resize();
+
+    proto::ReplicationRequestHeader hdr;
+    hdr.set_id(id());
+    hdr.set_type(proto::ReplicationRequestHeader::REQUEST);
+    hdr.set_management_type(proto::ReplicationManagementRequestType::REQUEST_STATUS);
+
+    _bufferPtr->serialize(hdr);
+
+    proto::ReplicationRequestStatus message;
+    message.set_id(id());
+    message.set_type(proto::ReplicationReplicaRequestType::REPLICA_FIND);
+
+    _bufferPtr->serialize(message);
+
+    send();
+}
+
+void FindRequestM::send() {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "send");
+
+    auto self = shared_from_base<FindRequestM>();
+
+    _messenger->send<proto::ReplicationResponseFind>(
+        worker(),
+        id(),
+        _bufferPtr,
+        [self] (std::string const& id,
+                bool success,
+                proto::ReplicationResponseFind const& response) {
+            self->analyze (success, response);
+        }
+    );
+}
+
+void FindRequestM::analyze(bool success,
+                           proto::ReplicationResponseFind const& message) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "analyze  success=" << (success ? "true" : "false"));
+    {
+        // This guard is made on behalf of an asynchronious callback fired
```

Comment:
IMO it would be kind to put a space after the log line and before the opening bracket to make it more obvious that the block is not related to the log macro. I also suggest putting a comment after the opening bracket `{  // context for lock`

Reply:
This code went through a significant refactoring due to a migration from macro LOCK to a more explicit RAII pattern. This has eliminated rough edges in this (and similar) file.